example-service/src/client.rs

```diff
@@ -38,20 +39,91 @@ async fn main() -> io::Result<()> {
         .parse::<SocketAddr>()
         .unwrap_or_else(|e| panic!(r#"--server_addr value "{}" invalid: {}"#, server_addr, e));
 
-    let name = flags.value_of("name").unwrap().into();
+    let num_connections: usize = flags
+        .value_of("num_connections")
+        .unwrap()
+        .parse()
+        .expect("num_connections needs an argument");
+
+    let (senders, receivers): (
+        Vec<tokio::sync::mpsc::Sender<Instant>>,
+        Vec<tokio::sync::mpsc::Receiver<Instant>>,
+    ) = (0..num_connections)
+        .map(|_| tokio::sync::mpsc::channel(ROUNDS + 1))
+        .unzip();
+
+    for mut sender in senders.into_iter() {
+        tokio::spawn(async move {
+            sender
+                .send(Instant::now())
+                .await
+                .expect("failed to send instant");
+            let transport = tarpc_bincode_transport::connect(&server_addr)
+                .await
+                .expect("failed to connect");
+            sender
+                .send(Instant::now())
+                .await
+                .expect("failed to send instant");
+
+            let mut client = service::WorldClient::new(client::Config::default(), transport)
+                .spawn()
+                .expect("failed to spawn client");
+
+            for _ in 0..ROUNDS {
+                client
+                    .hello(context::current(), NAME.into())
+                    .await
+                    .expect("failed to run hello");
+                sender
+                    .send(Instant::now())
+                    .await
+                    .expect("failed to send instant");
+            }
+        });
+    }
+
+    let mut latencies = Vec::new();
+    let mut instants = Vec::new();
+    for r in receivers.into_iter() {
+        let ivec: Vec<_> = r.collect().await;
+        for i in ivec.iter() {
+            instants.push(*i);
+        }
+        for (i1, i2) in ivec.iter().zip(&ivec[1..]) {
+            let dur = i2.duration_since(*i1);
+            latencies.push(dur);
```

Comment:
Am I correct that `latencies` contains the request times as well as the connect time? Should the connect times to be left out?